core/app/api/customer/groups/route.ts

```diff
@@ -0,0 +1,38 @@
+import { NextResponse } from 'next/server';
+import { z } from 'zod';
+
+export const CustomersGroup = z
+  .object({
+    id: z.number(),
+    name: z.string(),
+  })
+  .array();
+
+export type CustomerGroupType = z.infer<typeof CustomersGroup>;
+
+export async function GET() {
+  if (!process.env.BIGCOMMERCE_ACCESS_TOKEN) {
+    throw new Error('BIGCOMMERCE_ACCESS_TOKEN is not set');
+  }
+
+  const response = await fetch(
+    `https://api.bigcommerce.com/stores/${process.env.BIGCOMMERCE_STORE_HASH}/v2/customer_groups`,
+    {
+      method: 'GET',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+        'X-Auth-Token': process.env.BIGCOMMERCE_ACCESS_TOKEN,
+      },
+    },
+  );
+
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
```

Comment:
I don't think we want to throw an error here. Assuming this endpoint is being called from some frontend, we need return appropriate data which would include a message and http status code. If you wrap this whole function in try/catch and looking instance of generic `Error` as mentioned above this would work out.